### PR TITLE
Initial thoughts 

### PR DESCRIPTION
**Pull request by alexwebber**
Hey Eduardo,

Just some initial thoughts here, removing some duplication (e.g. code to validate account numbers) and taking advantage of some language features to reduce the lines of code and hopefully make it a little more understandable in places. What do you think?

Thanks,
Alex.

### CODE REVIEW

File: OCRManager/DigitHelper.cs [hunk 1]
```diff
@@ -66,10 +45,8 @@ public static IEnumerable<int> GetSimilarDigits(string input)
         {
             foreach (var candidateDigit in ValidDigits)
             {
-                if (1 == XORStrings(input, candidateDigit).Count(ch => ch == '1'))
-                {
+                if (input.Where((inputChar, index) => inputChar != candidateDigit[index]).Count() == 1)
```
Comment (eduherminio) [inline OCRManager/DigitHelper.cs:48]: Easy and simple!
I clearly overthought this after realizing that `.Intersect()` and `.Except()` didn't work due to the way digits were being stored.

File: OCRManager/Impl/OcrParser.cs [hunk 2]
```diff
@@ -1,98 +1,43 @@
...
-                new StringBuilder(_lineLength), new StringBuilder(_lineLength), new StringBuilder(_lineLength)
-            };
-
-            var inputInLines = input.Replace("\r", string.Empty).Split('\n');
-
-            for (int lineIndex = 1; lineIndex < inputInLines.Length; ++lineIndex)
-            {
-                if (inputInLines[lineIndex]?.Length == 0)
-                {
-                    inputInLines[lineIndex] = new string(' ', _lineLength);
```
Comment (eduherminio) [inline OCRManager/Impl/OcrParser.cs:42]: Tests are failing due to the removal of this operation.
Either this part needs to be re-added or tests need to be modified (adding a bunch of spaces to the first line of failing datasets, `111111111` and `444444444`).

**Review by eduherminio — CHANGES_REQUESTED** [review]
Either tests or parsing code need to be slightly modified for failing tests to pass, as described.

File: OCRManager/Impl/OcrParser.cs [hunk 3]
```diff
@@ -1,98 +1,43 @@
...
-        private void ParseNonFirstLine(string line, ICollection<StringBuilder> digits)
-        {
-            int counter = 0;
-            foreach (char ch in line)
+            for (var i = 0; i < 9; i++)
             {
-                counter = ParseCharacter(digits, counter, ch);
+                characters[i][0] = lines[0].Length > 0 && lines[0][(i * 3) + 1].Equals('_') ? '1': '0';
+                characters[i][1] = lines[1][(i * 3)].Equals('|') ? '1': '0';
+                characters[i][2] = lines[1][(i * 3) + 1].Equals('_') ? '1': '0';
```
Comment (eduherminio) [inline OCRManager/Impl/OcrParser.cs:38]: Beautiful, alternative approach!

File: OCRManager/Impl/OcrParser.cs (hunk 2 above)
Comment (alexwebber) [reply OCRManager/Impl/OcrParser.cs:42]: annoying, these all passed locally!

File: OCRManager/DigitHelper.cs (hunk 1 above)
Comment (alexwebber) [reply OCRManager/DigitHelper.cs:48]: That was exactly my initial thought process @ Intersect & Except!

**Review by eduherminio — APPROVED** [review]

**Pull request merged**